File: BaekJoon/function/1065.py
num = int(input())

count = 0

def isHan(a):
    if(a<100):
        return True
    while (a>=100):
        first = a%10
        second = (a%100)//10
        third = (a%1000)//100

        if(third-second != second-first):
            return False
        a//=10
        
    return True
        
for i in range(1,num+1):
    if(isHan(i)):
        count+=1

print(count)


    

File: BaekJoonSwift/BaekJoonSwift/Function/1065.swift
import Foundation

let n = Int(readLine()!)!
var result = 0
var isHan = true
for var i in 1...n {
    isHan = true
    if i/100 == 0 {
        result+=1
        continue
    }
    
    var first = i%10
    i/=10
    var second = i%10
    i/=10
    var third = i%10
    i/=10
    
    var diff = first - second
    while true {
        if second-third != diff {
            isHan = false
            break
        }
        if i == 0 {
            break
        }
        second = third
        third = i%10
        i/=10
    }
    
    if isHan {
        result+=1
    }
}

print(result)
